transform/v2/tsv-to-jsonl.py

```diff
@@ -0,0 +1,121 @@
+        ("organization", True, ["names", "shorts", "ids"]),
+        ("project", True, ["titles", "ids", "codes"]),
+    ]
+    if os.getenv("OMIT_JOURNAL"):
+        columns = [c for c in columns if c[0] != "journal"]
+    target_len = len(columns)
+    prev_i = None
+    count, err = 0, 0
+    try:
+        for i, row in enumerate(reader):
+            count += 1
+            if len(row) != target_len:
+                err += 1
+                print(f"Incorrect length. Line {i}: length {len(row)}", file=sys.stderr)
+            else:
+                row_errored = False
+                out = dict()
+                for j, col in enumerate(columns):
+                    col_name = col[0]
+                    is_list = col[1]
+                    try:
+                        named_struct = col[2]
+                    except IndexError:
+                        named_struct = None
+                    if list_sep in row[j] and not is_list:
+                        print(f"Possible array column {col_name}, in row {i} => {row[j].replace(list_sep, '<0002>')}",
+                              file=sys.stderr)
+                    if struct_sep in row[j] and named_struct is None:
+                        print(
+                            f"Possible struct column {col_name}, in row {i} => {row[j].replace(struct_sep, '<0003>')}",
+                            file=sys.stderr)
+                    if is_list:
+                        out_val = [v for v in row[j].split(list_sep) if not empty_or_n(v)]
+                        if named_struct is None:
+                            out[col_name] = out_val
+                        else:
+                            split_val = [v.split(struct_sep) for v in out_val]
+                            for k, field in enumerate(named_struct):
+                                try:
+                                    out[col_name + "_" + field] = [v[k] for v in split_val]
+                                except IndexError:
+                                    print(
+                                        f"IndexError {col_name}, in row {i} => {row[j].replace(struct_sep, '<0003>')}",
+                                        file=sys.stderr)
+                                    if not row_errored:
+                                        err += 1
+                                    row_errored = True
+                    else:
+                        out[col_name] = None if empty_or_n(row[j]) else row[j]
+                if not row_errored:
+                    print(json.dumps(out))
+            prev_i = i
+    except csv.Error as e:
+        print(e, file=sys.stderr)
+        print(f"{prev_i + 1}", file=sys.stderr)
+
+    print(f"{err}/{count}", file=sys.stderr)
+
+# @pytest.mark.parametrize("s,expected", [
+#     ("", True),
+#     ("   ", True),
+#     ("\\N", True),
+#     ("  \\N ", True),
+#     ("  foo ", False),
+# ])
+# def test_empty_or_n(s: str, expected: bool):
+#     assert empty_or_n(s) is expected
```

Comment:
Are those comments necessary?

Reply:
Necessary for now, as I use a bare python for processing and it doesn't have pytest installed then.
Nevertheless, I'll extract it to a separate file.